src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -617,6 +617,14 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             }
             break;
 
+            case NI_Vector64_ToVector128:
+            case NI_Vector64_ToVector128Unsafe:
+                if (op1Reg != targetReg)
```

Comment:
I don't believe we can use this optimization for `ToVector128` as it must ensure the upper bits are zeroed (unlike `ToVector128Unsafe` which can leave them non-deterministic).

Imagine a case such as:
```csharp
var x = Vector128.Create((byte)1);
x = x.GetLower().ToVector128();
```

In this scenario, the JIT could emit just the following:
```csharp
MOVI V0, #1
```

As we need the `MOVI` to generate the original value and `GetLower` can be a nop as the upper bits don't matter and so the register can be reused.
This logic allows the second `MOV` to then be elided as well, which would preserve a `Vector128` of all ones, rather than one which is half ones and half zeros.

Reply:
Yep, I think it was working for me was because `Vector128.GetLower()` was not optimized for arm64. Also after your comment, I noticed xarch implementation and that's what is done. I will fix it.

Reply:
Sadly we don't have test coverage for this kind of scenarios. :(

Reply:
Yes, I tried adding a test, but I don't see the wrong behavior as soon as use the result (`x` in your example). So I will skip adding test but while I was there, I also optimized `GetLower()` .